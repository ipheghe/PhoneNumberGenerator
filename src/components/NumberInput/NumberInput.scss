@import '../../common/functions.scss';

.input__container {
  display: flex;
  padding: 0;
  justify-content: center;
  align-items: center;

  .input__field {
    height: rem(54px);
    width: rem(100px);
    margin-right:0;
    border: rem(2px) #000 solid;
    padding: rem(5px) rem(12px);
    font-size: rem(20px);
  }

  .input__button {
    height: rem(60px);
    width: rem(250px);
    background-color: #57CCCB;
    color: #FFF;
    font-size: rem(20px);
    text-transform: capitalize;
    cursor: pointer;

    &:focus {
      outline:0;
    }
  }
}

@media screen and (max-width: rem(450px)) {
 .input__container {
    flex-direction: column;
    max-height: rem(120px);
    align-items: center;
    justify-content: center;

    .input__field {
      height: rem(54px);
      width: rem(100px);
      margin-bottom: rem(3px);
    }
  
    .input__button {
      height: rem(60px);
      width: rem(250px);
    }
  }
}
