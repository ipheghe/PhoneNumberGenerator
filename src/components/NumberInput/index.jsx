// react lib imports
import React from 'react';

// third-party-imports
import PropTypes from 'prop-types';

// styles
import './NumberInput.scss';

/**
 * The NumberInput component that renders an input and button element
 *
 * @returns {JSX} JSX
 */
const NumberInput = ({
  value,
  onChange,
  buttonText,
  handleButtonClick,
}) =>  (
  <div className="input__container">
    <input
      id="chosenNumber"
      className="input__field"
      type="number"
      name="chosenNumber"
      value={value}
      onChange={onChange}
      max="500"
      min="1"
    />
    <button
      className="input__button"
      type="button"
      onClick={handleButtonClick}
    >
      {buttonText}
    </button>
  </div>
);

NumberInput.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  onChange: PropTypes.func.isRequired,
  buttonText: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};

export default NumberInput;
