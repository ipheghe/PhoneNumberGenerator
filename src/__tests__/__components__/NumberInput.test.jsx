// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import render from 'react-test-renderer';

// components
import NumberInput from '../../components/NumberInput';

const props = {
  value: 4,
  onChange: jest.fn(),
  buttonText: 'generate',
  handleButtonClick: jest.fn(),
};

describe('NumberInput component', () => {
  it('should match component snapshot', () => {
    const tree = render.create(<NumberInput {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const shallowWrapper = shallow(<NumberInput {...props} />);
    expect(shallowWrapper).toBeDefined();
    expect(shallowWrapper.find('.input__container').length).toBe(1);
    expect(shallowWrapper.find('input').length).toBe(1);
    expect(shallowWrapper.find('button').length).toBe(1);
    expect(shallowWrapper.exists()).toBe(true);
  });
});
