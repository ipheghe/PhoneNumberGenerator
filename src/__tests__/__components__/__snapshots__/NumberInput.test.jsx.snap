// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput component should match component snapshot 1`] = `
<div
  className="input__container"
>
  <input
    className="input__field"
    id="chosenNumber"
    max="500"
    min="1"
    name="chosenNumber"
    onChange={[MockFunction]}
    type="number"
    value={4}
  />
  <button
    className="input__button"
    onClick={[MockFunction]}
    type="button"
  >
    generate
  </button>
</div>
`;
